pkg/smokescreen/smokescreen.go

```diff
@@ -895,14 +934,14 @@ func checkACLsForRequest(config *Config, req *http.Request, outboundHost string)
 
 	decision.role = role
 
-	submatch := hostExtractRE.FindStringSubmatch(outboundHost)
-	if len(submatch) < 2 {
+	// This host validation prevents IPv6 addresses from being used as destinations.
+	// Added for backwards compatibility.
+	if strings.ContainsAny(host, ":") {
```

Comment:
is this just because the regex failed on IPv6 addresses before?

Reply:
Yes. I hope to remove it in one of subsequent PRs as I hope to implement the IPv6 support. But for now I'd like to keep the current behaviour.